core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
 @homepage github.io
 -->
 
+<link rel="import" href="../polymer/polymer.html">
+
 <link rel="import" href="../core-media-query/core-media-query.html">
 <link rel="import" href="../core-selector/core-selector.html">
 
-<polymer-element name="core-drawer-panel" touch-action="auto">
-<template>
-
+<dom-module id="core-drawer-panel">
   <link rel="stylesheet" href="core-drawer-panel.css">
 
-  <core-media-query query="max-width: {{forceNarrow ? '' : responsiveWidth}}" queryMatches="{{queryMatches}}"></core-media-query>
+  <template>
+    <core-media-query query="[[mediaQuery]]" on-query-matches-changed="onQueryMatchesChanged"></core-media-query>
+
+    <core-selector class-name="[[coreSelectorClass]]" attrForSelected="id" selected="{{selected}}">
 
-  <core-selector class="{{ {'narrow-layout' : narrow, transition : transition, dragging : dragging, 'right-drawer': rightDrawer} | tokenList }}" valueattr="id" selected="{{selected}}">
+      <div id="main" style="[[mainStyle]]">
+        <content select="[main]"></content>
+        <div id="scrim" on-tap="[[togglePanel]]"></div>
+        <div id="edgeSwipeOverlay" class-name="[[swipeOverlayClass]]"></div>
+      </div>
 
-    <div id="main" _style="left: {{ narrow || rightDrawer ? '0' : drawerWidth }}; right: {{ rightDrawer ? (narrow ? '' : drawerWidth) : '' }};">
-      <content select="[main]"></content>
-      <div id="scrim" on-tap="{{togglePanel}}"></div>
-      <div id="edgeSwipeOverlay" hidden?="{{!narrow || disableEdgeSwipe}}"></div>
-    </div>
+      <div id="drawer" style="[[drawerStyle]]">
+        <content select="[drawer]"></content>
```

Comment:
I see this as a good use case even though the `select` attribute uses CSS selector syntax. I would prefer to prefix these with `data-` though since `drawer` and `main` are invalid attributes on native elements.

```html
<core-drawer-panel>
  <div data-main></div>
  <div data-drawer></div>
</core-drawer-panel>
```